cg/clients/chanjo2/api.py

```diff
@@ -0,0 +1,40 @@
+"""Coverage analysis API for human clinical sequencing data."""
+
+import logging
+from typing import Any
+
+import requests
+
+from cg.clients.chanjo2.models import CoverageData, CoverageRequest
+
+LOG = logging.getLogger(__name__)
+
+
+class Chanjo2APIClient:
+    """
+    Chanjo2 API to communicate with a d4tools software in order to return coverage
+    over genomic intervals.
+    """
+
+    def __init__(self, config: dict[str, str]):
+        self.headers = {
+            "Content-Type": "application/json",
+            "accept": "application/json",
+        }
+        self.host = config["chanjo2"]["host"]
+
+    def get_coverage(self, coverage_request: CoverageRequest) -> CoverageData | None:
+        """Send a POST request to the coverage endpoint to retrieve gene coverage summary data."""
+        endpoint: str = f"{self.host}/coverage/d4/genes/summary"
+        post_data: dict[str, Any] = coverage_request.model_dump()
+        try:
+            response = requests.post(url=endpoint, headers=self.headers, json=post_data)
```

Comment:
I guess it also depends on the chanjo API, if it is built wrong then it could be the case that it has to be a post request 🤢 If that is the case and we maintain the chanjo API, we should patch it to be a get request.

Reply:
It sounds inefficient to send such a list. Would it be possible to send a panel id instead? This is definitely out of scope and it sounds like there is a lot to clean up in https://github.com/Clinical-Genomics/chanjo2

Reply:
I'll start creating issues in this repo, we can have stuff like this going on

Reply:
I'll create issues in that repo to track these problems

Reply:
That might be complicated since these panels are specific to our use case. But starting a discussion with an issue could be a good approach. Thanks!